pkg/cloud/instance.go

```diff
@@ -275,15 +275,17 @@ func (c *client) GetOrCreateVMInstance(
 		listVirtualMachineParams.SetZoneid(fd.Spec.Zone.ID)
 		listVirtualMachineParams.SetNetworkid(fd.Spec.Zone.Network.ID)
 		listVirtualMachineParams.SetName(csMachine.Name)
-		if listVirtualMachinesResponse, err2 := c.cs.VirtualMachine.ListVirtualMachines(listVirtualMachineParams); err2 == nil && listVirtualMachinesResponse.Count > 0 {
-			csMachine.Spec.InstanceID = pointer.StringPtr(listVirtualMachinesResponse.VirtualMachines[0].Id)
-		} else {
+		listVirtualMachinesResponse, err2 := c.cs.VirtualMachine.ListVirtualMachines(listVirtualMachineParams)
+		if err2 != nil || listVirtualMachinesResponse.Count <= 0 {
 			c.customMetrics.EvaluateErrorAndIncrementAcsReconciliationErrorCounter(err2)
+			return err
 		}
-		return err
+		csMachine.Spec.InstanceID = pointer.StringPtr(listVirtualMachinesResponse.VirtualMachines[0].Id)
+		csMachine.Status.InstanceState = listVirtualMachinesResponse.VirtualMachines[0].State
+	} else {
```

Comment:
I think the rationale for lifting the scope of deployVMResp out of the if was so that this else could be eliminated (i.e., the code within it be move into the main-line flow).

Reply:
My bad here, I misread the indentation in the github diff.  Understood now that the else block I called out is not the one Max asked be removed, and that it need to stand.